include/deal.II/base/tensor_accessors.h

```diff
@@ -483,7 +484,7 @@ namespace TensorAccessors
     // subsequently stored indices:
 
     template <int rank, typename S>
-    class StoreIndex : private S
+    class StoreIndex : public S
```

Comment:
The inheritance lost its purpose some iterations of this class ago. An artifact. I remove it.